### DISCUSSION TITLE
Where is the Encryption Key located?

### DISCUSSION THREAD

**Discussion by xKugeki** [dani-garcia/vaultwarden#3555] [Q&A] [answered]
Where is the location for the encryption key? (for the database encryption) 
And is there any way to use an own encryption key?

Have a nice day :)

**Comment by RealOrangeOne** [comment]
There isn't really "an" encryption key stored anywhere - it's derived from a number of different pieces of information. See the [Bitwarden whitepaper](https://bitwarden.com/help/bitwarden-security-white-paper/) for details on how things work behind the scenes.

As for using your own encryption key, both because of the implementation above, and because there's no real value add, no that's not possible. If you're after a password manager that uses an encryption key you directly control, you'll need to look elsewhere.

**Reply by BlackDex** [reply, marked as answer]
To clarify it a bit more.
All data send from and to Vautlwarden is encrypted, except for a few fields like login email, organization name, but that is about it.
All other fields are encrypted and decrypted on the client side. The server has no way to decrypt the vault it self, since it does not have the needed information to do so.
